docs/fundamentals/syslib-diagnostics/syslib1039.md

```diff
@@ -0,0 +1,15 @@
+---
+title: SYSLIB1039 error
+description: Learn about the diagnostic that generates compile-time error SYSLIB1039.
+ms.date: 10/26/2023
+---
+
+# SYSLIB1039: JsonSourceGenerator encountered a `JsonDerivedTypeAttribute` annotation with `JsonSourceGenerationMode.Serialization` enabled
+
+<xref:System.Text.Json.Serialization.JsonDerivedTypeAttribute> is not supported in source-generation mode.
```

Comment:
`JsonDerivedTypeAttribute` annotations _are_ supported in source gen, however they do not work for contexts annotated with `JsonSourceGenerationMode.Serialization` (i.e. it works with the metadata source generator, but not in fast-path methods. Hence contexts configured to only use fast-path can't support this).